docs/advanced/api_examples.yml

```diff
@@ -37,3 +43,103 @@
     {
         "flocker": "0.4.0"
     }
+
+-
+  id:
+    "create dataset"
+
+  doc: |
+    Specifing as few parameters as possible, create a new dataset on the
+    cluster.  A node is specified to have the primary manifestation for the new
+    dataset.  The unique dataset identifier is automatically generated by the
+    server and returned with the response.
+
+  request: |
+    POST /v1/datasets HTTP/1.1
+
+    {"primary": "%(NODE_0)s"}
+
+  response: |
+    HTTP/1.1 201 Created
+
+    {"dataset_id": "a5f75af7-3fb9-4c1a-81ce-efeeb9f2c788", "primary": "%(NODE_0)s", "metadata": {}}
```

Comment:
As noted for version endpoint example, this isn't actually accurate representation of output until https://clusterhq.atlassian.net/browse/FLOC-1254 is done.